cypress/integration/checkout/checkout.basic.spec.js

```diff
@@ -1,54 +1,122 @@
-        cy.getUserServicePoints(Cypress.env('users')[0].id);
-
-        cy.createInstance({
+        InventoryInstances.createFolioInstanceViaApi({
           instance: {
-            instanceTypeId: Cypress.env('instanceTypes')[0].id,
-            title: `Pre-checkout instance ${Number(new Date())}`,
+            instanceTypeId: itemData.instanceTypeId,
+            title: itemData.instanceTitle,
           },
-          holdings: [{
-            holdingsTypeId: Cypress.env('holdingsTypes')[0].id,
-            permanentLocationId: Cypress.env('locations')[0].id,
-            sourceId: source.id,
-          }],
+          holdings: [
+            {
+              holdingsTypeId: itemData.holdingTypeId,
+              permanentLocationId: defaultLocation.id,
+            },
+          ],
           items: [
-            [{
-              barcode: ITEM_BARCODE,
-              missingPieces: '3',
-              numberOfMissingPieces: '3',
+            {
+              barcode: itemData.barcode,
               status: { name: 'Available' },
-              permanentLoanType: { id: Cypress.env('loanTypes')[0].id },
-              materialType: { id: Cypress.env('materialTypes')[0].id },
-            }],
+              permanentLoanType: { id: itemData.loanTypeId },
+              materialType: { id: itemData.materialTypeId },
+            },
           ],
         });
+      })
+      .then((specialInstanceIds) => {
+        itemData.testInstanceIds = specialInstanceIds;
       });
-  });
 
-  it('Basic flow', function () {
-    cy.visit('/checkout');
+    cy.createTempUser(
+      [
+        permissions.checkinAll.gui,
+        permissions.uiCirculationViewCreateEditDelete.gui,
+        permissions.uiUserCreate.gui,
+        permissions.inventoryAll.gui,
+        permissions.checkoutCirculatingItems.gui,
+      ],
+      userData.group
+    )
+      .then((userProperties) => {
+        userData.personal.lastname = userProperties.username;
+        userData.password = userProperties.password;
+        userData.userId = userProperties.userId;
+        userData.barcode = userProperties.barcode;
+        userData.firstName = userProperties.firstName;
+      })
+      .then(() => {
+        UserEdit.addServicePointViaApi(servicePoint.id, userData.userId, servicePoint.id);
 
-    cy.checkOutItem(Cypress.env('users')[0].barcode, ITEM_BARCODE);
+        cy.login(userData.personal.lastname, userData.password, { path: TopMenu.checkOutPath, waiter: CheckOutActions.waitLoading });
+      });
+  });
+
+  after('Delete New Service point, Item and User', () => {
+    CheckInActions.checkinItemViaApi({
+      itemBarcode: itemData.barcode,
+      servicePointId: servicePoint.id,
+      checkInDate: new Date().toISOString(),
+    });
+    InventoryInstances.deleteInstanceAndHoldingRecordAndAllItemsViaApi(itemData.barcode);
+    UserEdit.changeServicePointPreferenceViaApi(userData.userId, [servicePoint.id]);
+    Users.deleteViaApi(userData.userId);
```

Comment:
Actually we need to create two users according to the preconditions. One user should create another user (Created test active user with a username and barcode exist) and checkout the item to that user.